jwst/extract_1d/soss_extract/soss_boxextract.py

```diff
@@ -98,17 +98,25 @@ def box_extract(scidata, scierr, scimask, box_weights, cols=None):
     mask = scimask[:, cols].copy()
     box_weights = box_weights[:, cols].copy()
 
-    # Check that all invalid values are masked.
-    if not np.isfinite(data[~mask]).all():
+    # Check that all invalid values in the extraction region are masked.
+    extract_region = (box_weights > 0)
+    condition = extract_region & ~mask
+
+    if not np.isfinite(data[condition]).all():
         message = 'scidata contains un-masked invalid values.'
         log.critical(message)
         raise ValueError(message)
 
-    if not np.isfinite(error[~mask]).all():
+    if not np.isfinite(error[condition]).all():
         message = 'scierr contains un-masked invalid values.'
         log.critical(message)
         raise ValueError(message)
 
+    # Set all pixels values outside of extraction region to Nan
+    # so it will be correctly handle by np.nansum.
+    data = np.where(extract_region, data, np.nan)
+    error = np.where(extract_region, error, np.nan)
```

Comment:
@AntoineDarveau found a bug in the boxextract code which I think exists in your latest branch as well (not to say that you wrote it - quite possible it filtered in from merging some of my commits!) Just wanted to warn you, as it prevents any data output.